lms/djangoapps/certificates/tests/test_queue.py

```diff
@@ -110,18 +136,33 @@ def assert_queue_response(self, mode, expected_mode, expected_template_name):
                 mock_send.return_value = (0, None)
                 self.xqueue.add_cert(self.user_2, self.course.id)
 
-        # Verify that the task was sent to the queue with the correct callback URL
-        self.assertTrue(mock_send.called)
-        __, kwargs = mock_send.call_args_list[0]
-
-        actual_header = json.loads(kwargs['header'])
-        self.assertIn('https://edx.org/update_certificate?key=', actual_header['lms_callback_url'])
-        certificate = GeneratedCertificate.objects.get(user=self.user_2, course_id=self.course.id)
+        # Ensure that a certificate record has been created, and
```

Comment:
I'm not sure it makes sense for an assertion helper to be responsible for building up the state that is being asserted -- at least that's what appears to be happening in this method compared to its name.